### ISSUE TITLE
define split differently for DR methods

### ISSUE THREAD

**Issue by MattScicluna** [latent-reasoning-works/manylatents#24]
For our non-neural network based methods, our metrics should contain points fitted on.
So the test set should also contain points in training set.
adding another split is a potential solution.
